File: removeDuplicates.py
def removeDuplicates(nums):
    if len(nums) == 0:
        return []
    
    unique = [nums[0]]  # start with first element
    
    for i in range(1, len(nums)):
        if nums[i] != nums[i - 1]:
            unique.append(nums[i])  # add only if not duplicate
    
    return unique

print(removeDuplicates([1, 1, 2, 2, 3]))

File: Javascript/Array/removeduplicates.js
const removeDuplicates = (nums) => {
  if (nums.length === 0) return [];

  const unique = [nums[0]]; // first element is always unique

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] !== nums[i - 1]) {
      unique.push(nums[i]);
    }
  }

  return unique;
};

console.log(removeDuplicates([1, 1, 2, 2, 3]));
